packages/store/src/utils/selector-utils.ts

```diff
@@ -5,24 +5,25 @@ import { memoize } from '../utils/memoize';
  * Function for creating a selector
  * @param selectors The selectors to use to create the arguments of this function
  * @param originalFn The original function being made into a selector
+ * @param creationMetadata
  */
-export function createSelector(
-  selectors: any[],
-  originalFn: any,
+export function createSelector<T extends (...args: any[]) => R, R>(
+  selectors: any[] | undefined,
+  originalFn: T,
```

Comment:
I'm hopeing that this improves the overall type recognition for selectors.
Nice work!